### PR TITLE
Initial implementation of a dedicated fixed-form parser

### PR DESCRIPTION
**Pull request by certik**
This PR implements a dedicated fixed-form tokenizer that uses a recursive descent parser to properly understand the fixed form source code and it emits tokens that are correct. We then use the existing Bison based parser to parse it to AST.

This is just the initial version, which passes all tests in master and adds a few more. We will then iteratively improve this to parse all fixed-form code.

### CODE REVIEW

**Comment by certik** [comment]
TODO:
* [ ] tokenizing floats (yysval)
* [ ] remove debugging prints
* [ ] there was some issue with compiling to binary
* [ ] just pass the same tests as in `main`

**Comment by certik** [comment]
@konradha the last commit fixes error locations for syntax errors, such as:
```
syntax error: Syntax error in the fixed-form parser
  --> tests/fixed_form1.f:58:8
   |
58 |       1abserr = 0.0d+00
   |        ^^^^^^ 
```

That works well, except if it is inside a continuation line, where it is shifted by 2 characters, don't know why yet.

File: src/bin/lfortran.cpp [hunk 1]
```diff
@@ -1358,8 +1359,11 @@ EMSCRIPTEN_KEEPALIVE char* emit_wasm_from_source(char *input) {
 
 #endif
 
+
 int main(int argc, char *argv[])
 {
+    // extern int yydebug;
+    // yydebug=1;
```
Comment (certik) [inline src/bin/lfortran.cpp:1366]:
```suggestion
```

File: src/lfortran/parser/parser.cpp [hunk 2]
```diff
@@ -190,6 +223,9 @@ void copy_rest_of_line(std::string &out, const std::string &s, size_t &pos)
             skip_rest_of_line(s, pos);
             out += '\n';
             return;
+        } else if (s[pos] == ' ') {
+            // Skip white space in a fixed-form parser
+            pos++;
```
Comment (certik) [inline src/lfortran/parser/parser.cpp:228]: I think this change causes the shift by a few characters in the error messages.

**Comment by certik** [comment]
@konradha I fixed the locations. I think we have it.

File: src/lfortran/parser/parser.cpp (hunk 2 above)
Comment (certik) [reply src/lfortran/parser/parser.cpp:229]: Fixed.

**Comment by certik** [comment]
@konradha  done!!

**Comment by konradha** [comment]
@certik this is great, thanks for pushing so hard today!!

**Pull request merged**